### DISCUSSION TITLE
Error handling

### DISCUSSION THREAD

**Discussion by niwsa** [sergiodxa/remix-auth-oauth2#16] [Q&A] [answered]
I have been using this strategy to implement [remix-auth-saml](https://github.com/boxyhq/remix-auth-saml). Couple of questions I have  
1. Is it intentional to throw 401 here rather than catching it with the `this.failure` method like it's done for `verify` ? https://github.com/sergiodxa/remix-auth-oauth2/blob/7857211630aa136213dfbe83560aa2eb59563b1a/src/index.ts#L331 
2. Should the callback routes (e.g. app/routes/auth/github/callback.tsx) export a default component so that it's treated as a non-resource route and any error gets caught in the app root CatchBoundary ? Unless I put something like below the CatchBoundary is not shown.
```js
export default function Render() {
  return null;
}
```

**Comment by sergiodxa** [comment, marked as answer]
That's not correct, it should use `this.failure` instead.
